src/main/scala/edu/berkeley/ce/rockslicing/SeedJointSelector.scala

```diff
@@ -0,0 +1,257 @@
+      throw new IllegalArgumentException(s"Error: Not enough joints to generate required number of seed joints. ${jointSet.length} joints but"+
+        s" $numProcessors processors. Try using fewer processors.")
+    }
+
+    val totalVolume = rockVolume.volume
+    val volumePerProc = rockVolume.volume / numProcessors
+    val seedJoints = selectSeedJoints(jointSet, rockVolume, volumePerProc, totalVolume,
+      Seq.empty[Joint], tolerance, numProcessors)
+
+    if (stepSize < 0.01) {
+      // Unable to find satisfactory seed joints using current joint set
+      None
+    } else if (seedJoints.length != numProcessors - 1) {
+      val newStepSize = if (tolerance >= 1.0) { stepSize / 2.0 } else { stepSize }
+      val newTolerance = if (tolerance >= 1.0) { newStepSize } else { tolerance + stepSize }
+      findSeedJoints(jointSet, rockVolume, numProcessors, newTolerance, newStepSize)
+    } else {
+      val initialBlock = Seq(rockVolume)
+      val blocks = seedJoints.foldLeft(initialBlock) { (currentBlocks, joint) =>
+        currentBlocks.flatMap(_.cut(joint))
+      }
+      val nonRedundantBlocks = blocks map { case block @ Block(blockCenter, _, _) =>
+          Block(blockCenter, block.nonRedundantFaces)
+      }
+      val volumes = nonRedundantBlocks.map(_.volume)
+      val avgVolume = volumes.sum / volumes.length
+      println("\nLOAD BALANCE INFORMATION:")
+      println(s"Average Volume: %.2f".format(avgVolume))
+      println(s"Max Volume: %.2f".format(volumes.max))
+      println(s"Min Volume: %.2f".format(volumes.min))
+      Some(seedJoints)
+    }
+  }
+
+  /**
+    * Iterates through input joint set and selects joints that divide the initial block into approximately
+    * equal volumes
+    *
+    * @param jointSet Seq of Joints representing a persistent joint set
+    * @param initialBlock Block that should be subdivided by seed joints into approximately equal volumes
+    * @param volumePerProcessor Ideal volume per processor for load balance
+    * @param totalVolume Total volume of rock volume in analysis
+    * @param selectedJoints Seq of Joints that have been identified as seed joints
+    * @param tolerance Acceptable deviation from ideal volume per processor
+    * @param numProcessors Number of processors that will be used in the analysis
+    * @return Seq of joints that divide initial block into approximately equal volumes
+    */
+  @tailrec
+  private def selectSeedJoints(jointSet: Seq[Joint], initialBlock: Block, volumePerProcessor: Double,
+                            totalVolume: Double, selectedJoints: Seq[Joint], tolerance: Double,
+                            numProcessors: Int): Seq[Joint] = {
+    if (selectedJoints.length == numProcessors - 1) {
+      // Enough seed joints have been found
+      selectedJoints
+    } else if (jointSet.length > 1) {
+      // More than 1 joint in input joint set
+      val jointOption = testVolumes(jointSet.head, jointSet.tail.head, initialBlock,
```

Comment:
Let's write `jointSet(0)` and `jointSet(1)` instead of using `.head` and `.tail.head`.